nservicebus/transports/transactions.md

```diff
@@ -51,6 +51,22 @@ In this mode handlers will execute inside the `TransactionScope` created by the
 
 Distributed transactions do not guarantee atomicity for an outside observer. For example outgoing messages might be dispatched and processed by some other endpoint before changes to the database get committed. However it is guaranteed that eventually all resources will sync up to reflect the outcome of the transaction.
 
+Consider a message exchange scenario with a saga that models a simple order lifecycle:
+
+1. `OrderSaga` receives a `StartOrder` message
+1. New `OrderSagaData` instance is created and stored in RavenDB.
+1. At this point RavenDB transaction is prepared but not committed.
+1. `OrderSaga` sends `VerifyPayment` message to `PaymentService`.
+1. At this point MSMQ transactions (send and receive) are prepared but not yet committed.
+1. NServiceBus completes the distributed transaction and the DTC orders MSMQ and RavenDB to commit their local transactions.
+1. MSMQ is a local resource and commits very fast. `StartOrder` message is removed from the input queue and `VerifyPayment` is immediately sent to `PaymentService`.
+1. At this point RavenDB transaction received order to commit but has not yet persisted the `OrderSagaData` instance.
+1. `PaymentService` receives `VerifyPayment` message and immediately responds with `CompleteOrder` message to the originating `OrderSaga`.
+1. `OrderSaga` receives the `CompleteOrder` message and attempts to complete the saga.
+1. At this point RavenDB has still not persisted the `OrderSagaData` instance and the `OrderSaga` fails to find an instance of the saga to complete.
+
+In the example above atomicity is preserved from the `OrderSaga` perspective: receiving `StartOrder` message, storing `OrderSagaData` in RavenDB and sending `VerifyPayment` message are one atomic operation. State of the system is not consistent for an outside observer such as the `PaymentService` and results in unexpected behavior: `CompleteOrder` message is sent even though the `OrderSagaData` does not yet exist in RavenDB and cannot be completed.
+
```

Comment:
I would add something about solution: locks. When the storage maintains the locks by itself (SQL Server) nothing needs to be done. When the storage does not (RavenDB, Oracle), marking a message as starting saga will enforce the lock conflict on the SagaID.